### PR TITLE
Remove exclude voted proposals feature

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

Remove "Show only proposals you can still vote for" checkbox. This is the first step to get rid of `ENABLE_VOTING_INDICATION` feature flag.

# Changes

- Remove the `<Checkbox>`  from `NnsProposalsFilters` component.
- Remove obsolete code from `ProposalsFiltersStore`:
  - Remove `excludeVotedProposals` (reflects the removed checkbox state).
  - Remove `lastAppliedFilter` (it was used only to render a preloader when `excludeVotedProposals` is checked).
  - Removed `reload()` that was there to fix — “The filter to show only the ones you can vote on was being used even though the user was not logged in” (#1562).
- Remove `isExcludedVotedProposal` util.
- Remove unused arguments because neurons and identity were needed only for `isExcludedVotedProposal()`.

# Tests

- Remove `getVotableProposalsOnlyCheckboxPo`.
- Update tests.

# Todos

- [ ] Add entry to changelog (if necessary).
Not necessary.

### CODE REVIEW

File: frontend/src/lib/derived/proposals.derived.ts [hunk 1]
```diff
@@ -45,19 +42,13 @@ export const sortedProposals: Readable<ProposalsStore> = derived(
 const hide = ({
```
Comment (dskloetd) [inline frontend/src/lib/derived/proposals.derived.ts:42]: Why does this function exist? Can `hideProposal` be used instead?
Do you understand what the comment above (from `HACK:`) is saying? Is it relevant to what you are changing?

File: frontend/src/lib/utils/proposals.utils.ts [hunk 2]
```diff
@@ -78,25 +76,14 @@ export const getNnsFunctionKey = (
...
-  neurons,
-  identity,
 }: {
   proposalInfo: ProposalInfo;
   filters: ProposalsFiltersStore;
-  neurons: NeuronInfo[];
-  identity: Identity | undefined | null;
-}): boolean =>
-  !matchFilters({ proposalInfo, filters }) ||
-  (!get(ENABLE_VOTING_INDICATION) &&
```
Comment (dskloetd) [inline frontend/src/lib/utils/proposals.utils.ts:96]: Removing this doesn't really belong in this PR, right?

Comment (mstrasinskis) [reply frontend/src/lib/utils/proposals.utils.ts:96]: It depends on the point of view, but I got your point.

File: frontend/src/lib/derived/proposals.derived.ts (hunk 1 above)
Comment (mstrasinskis) [reply frontend/src/lib/derived/proposals.derived.ts:42]: Good catch! 
1. There was more logic before.
2. Yes, I'll do the refactoring.
3. Yes.
4. Kind of.

File: frontend/src/lib/utils/proposals.utils.ts (hunk 2 above)
Comment (mstrasinskis) [reply frontend/src/lib/utils/proposals.utils.ts:96]: Actually I still think that it belongs to this PR, because there is nothing besides `isExcludedVotedProposal` which was remove.

Comment (dskloetd) [reply frontend/src/lib/utils/proposals.utils.ts:96]: In the diff that I see it changes from
```
  !matchFilters({ proposalInfo, filters }) ||
  (!get(ENABLE_VOTING_INDICATION) &&
    nonNullish(identity) &&
    !identity.getPrincipal().isAnonymous() &&
    isExcludedVotedProposal({ proposalInfo, filters, neurons }));
 ```
to just
```
!matchFilters({ proposalInfo, filters });
```
Is that not the change you made?

Comment (dskloetd) [reply frontend/src/lib/utils/proposals.utils.ts:96]: Oh, now I realized that because of the `&&` it all collapses to false...

**Review by dskloetd — APPROVED** [review]
